src/components/ui/SkuSelector/skuVariants.ts

```diff
@@ -0,0 +1,71 @@
+import type { NextRouter } from 'next/router'
+
+export type SkuVariantsByName = Record<
+  string,
+  Array<{ alt: string; src: string; label: string; value: string }>
+>
+
+export function getSkuSlug(
+  slugsMap: Record<string, string>,
+  selectedVariations: Record<string, string>,
+  dominantVariation: string
+) {
+  let slugsMapKey = ''
+
+  for (const key in selectedVariations) {
+    if (key in selectedVariations) {
+      const variationValue = selectedVariations[key]
+
+      slugsMapKey += `${key}-${variationValue}-`
+    }
+  }
+
+  // Remove trailing '-'
+  slugsMapKey = slugsMapKey.slice(0, slugsMapKey.length - 1)
+
+  const variantExists = slugsMapKey in slugsMap
+
+  if (!variantExists) {
+    const possibleVariants = Object.keys(slugsMap)
+    const firstVariationForDominantValue = possibleVariants.find((slug) =>
+      slug.includes(
+        `${dominantVariation}-${selectedVariations[dominantVariation]}`
+      )
+    )
+
+    return slugsMap[firstVariationForDominantValue ?? possibleVariants[0]]
+  }
+
+  return slugsMap[slugsMapKey]
```

Comment:
I think we can simplify this, what about:

```suggestion
   const slugsMapKey = Object.entries(selectedVariations).flat().join('-')

  if (slugsMapKey in slugsMap) return slugsMap[slugsMapKey]

  const possibleVariants = Object.keys(slugsMap)
  const firstVariationForDominantValue = possibleVariants.find((slug) =>
    slug.includes(
      `${dominantVariation}-${selectedVariations[dominantVariation]}`
    )
  )

  return slugsMap[firstVariationForDominantValue ?? possibleVariants[0]]
```

Reply:
Awesome!!